File: cases/taint/ruby/dispatch-table-positive/dispatch_table.rb
def dfb_source # DFB-SOURCE: dispatch-table-input
  "tainted"
end

def dfb_sink(value) # DFB-SINK: dispatch-table-sink
end

def run
  table = {
    "leak" => ->(value) { dfb_sink(value) },
    "drop" => ->(value) { dfb_sink("clean") }
  } # DFB-WITNESS: dispatch-table-build
  key = "leak"
  table[key].call(dfb_source)
end

File: cases/taint/python/dispatch-table-positive/dispatch_table.py
def dfb_source():  # DFB-SOURCE: dispatch-table-input
    return "tainted"


def dfb_sink(value):  # DFB-SINK: dispatch-table-sink
    pass


def leak(value):
    dfb_sink(value)


def drop(value):
    dfb_sink("clean")


def run():
    table = {"leak": leak, "drop": drop}  # DFB-WITNESS: dispatch-table-build
    key = "leak"
    table[key](dfb_source())
